cannon/mipsevm/memory/page.go

```diff
@@ -124,3 +124,35 @@ func (p *CachedPage) MerkleizeSubtree(gindex uint64) [32]byte {
 	}
 	return p.Cache[gindex]
 }
+
+func (p *CachedPage) MerkleizeNode(addr Word, gindex uint64) [32]byte {
+	_ = p.MerkleRoot() // fill cache
+	if gindex >= PageSize/32 {
+		if gindex >= PageSize/32*2 {
+			panic("gindex too deep")
+		}
+
+		// it's pointing to a bottom node
+		nodeIndex := gindex & (PageAddrMask >> 5)
+		return *(*[32]byte)(p.Data[nodeIndex*32 : nodeIndex*32+32])
+	}
+	return p.Cache[gindex]
+}
+
+func (p *CachedPage) GenerateProof(addr Word) [][32]byte {
```

Comment:
This seems to only be used by the MPT implementation from asterisc so could be removed along with it.